API/models/Posts.js

```diff
@@ -0,0 +1,177 @@
+const mongoose = require('mongoose');
+const Schema = mongoose.Schema;
+
+// TODO: implement comments model
+// require('./Comments');
+// const Comment = mongoose.model('Comment');
+
+const PostSchema = new Schema({
+	author: {
+    //type: Schema.Types.ObjectId,
+    type: String,
+    required: true,
+    ref: 'User'
+  },
+  subscribedUsers: {
+    type: Array,
+    default: [],
+    ref: 'User'
+  },
+	content: {
+		type: String,
+		required: true,
+  },
+  image: {
+    type: Buffer,
+  },
+	tags: {
+		type: Array,
+		required: true,
+  },
+  comments: {
+    type: Array,
+    ref: 'Comment'
+  },
+  createdAt: {
+    type: Object,
+  },
+  updatedAt: {
+    type: Object,
+  }
+});
+
+PostSchema.statics.create = function(post) {
+  // Implement a character limit client-side
+  if (!post.createdAt) {
+    post.createdAt = new Date();
+  }
+	var formattedTags = [];
+	post.tags.forEach((tag) => {
+    formattedTags.push(tag.toLowerCase());
+  });
+  var post = {
+    ...post,
+    subscribedUsers: post.author,
+    tags: formattedTags,
+  }
+	return new Promise((resolve, reject) => {
+    this.find({ content: post.content }).then(existingPost => {
+      if (existingPost && existingPost.length) {
+        reject(`Hmm, it looks like a post with the exact same content already exists. Try something new!`);
+      } else {
+        const newPost = new this(post);
+        newPost.save().then(post => {
+          resolve(post);
+        })
+        .catch(err => {
+          reject(err);
+        });
+      }
+    })
+    .catch(err => {
+      reject(err);
+    });
+  });
+}
+
+PostSchema.statics.get = function(id) {
+  if (typeof id === 'string') id = mongoose.Types.ObjectId(id);
+
+  return new Promise((resolve, reject) => {
+    this.findById(id).then(post => {
+      if (post) {
+        resolve(post);
+      }
+      else {
+        reject(`Couldn't find a post with that ID`);
+      }
+    })
+    .catch(err => {
+      reject(err);
+    });
+  })
+}
+
+PostSchema.statics.updatePost = function(id, postObj) {
+  if (typeof id === 'string') id = mongoose.Types.ObjectId(id);
+
+  postObj = {
+    ...postObj,
+    updatedAt: new Date()
+  }
+
+  return new Promise((resolve, reject) => {
+    this.update({ _id: id }, postObj).then(post => {
```

Comment:
That makes sense to me, we definitely should only be updating comments through the defined endpoints. We could probably just filter the hashmap to the keys we want through lodash's `.pick` method instead of manually updating fields.